elastic/kibana
# [Fleet] Ensure that Elastic Agent is installed when updating policy with monitoring_enabled

Fixes https://github.com/elastic/kibana/issues/129863
## Summary

Ensure that Elastic Agent integration gets installed when a policy is updating with `monitoring_enabled`. This would prevent cases like the bug occurring, where the Elastic Agent is not installed so no logs would ever appear.

#### Testing

- Create an agent policy **without** monitoring enabled
- Enroll an agent in the policy
- Verify elastic agent package is not installed
- In Logs UI, no logs are visible:

![Screenshot 2022-06-03 at 15 03 44](https://user-images.githubusercontent.com/16084106/171874765-ff977841-8d9b-4e45-a5a0-60259b5a699b.png)

- Update the agent policy to have monitoring enabled. After updating:

1. Verify that Elastic Agent is installed
2. Check logs UI dropdown - the datasets should all be available and containing logs

![Screenshot 2022-06-03 at 15 05 08](https://user-images.githubusercontent.com/16084106/171874893-ffd8ca8a-a58f-4060-aff8-ab18152c8b3e.png)

### Checklist
